internal/common/utils.go

```diff
@@ -86,5 +100,38 @@ func SendEvent(event *dtos.Event, correlationID string, dic *di.Container) {
 		} else {
 			lc.Debugf("Event(profileName: %s, deviceName: %s, sourceName: %s, id: %s) pushed to Coredata", event.ProfileName, event.DeviceName, event.SourceName, event.Id)
 		}
+		sent = true
+	}
+
+	if sent {
+		handleEventReadingMetrics(event, lc, dic)
+	}
+}
+
+func handleEventReadingMetrics(event *dtos.Event, lc logger.LoggingClient, dic *di.Container) {
+	// Just need to check one of the metrics
+	if eventsSent == nil {
```

Comment:
it looks like the race condition might happen in this function

Reply:
good point. Will refactor to add metrics initialization function to avoid using mutex.

Reply:
Refactored to initialize the metrics during bootstrapping.